cmd/platform/sampledata.go

```diff
@@ -0,0 +1,628 @@
+	if err != nil || channelsPerTeam < 0 {
+		return errors.New("Invalid channels-per-team parameter")
+	}
+	users, err := cmd.Flags().GetInt("users")
+	if err != nil || users < 0 {
+		return errors.New("Invalid users parameter")
+	}
+	teamMemberships, err := cmd.Flags().GetInt("team-memberships")
+	if err != nil || teamMemberships < 0 {
+		return errors.New("Invalid team-memberships parameter")
+	}
+	channelMemberships, err := cmd.Flags().GetInt("channel-memberships")
+	if err != nil || channelMemberships < 0 {
+		return errors.New("Invalid channel-memberships parameter")
+	}
+	postsPerChannel, err := cmd.Flags().GetInt("posts-per-channel")
+	if err != nil || postsPerChannel < 0 {
+		return errors.New("Invalid posts-per-channel parameter")
+	}
+	directChannels, err := cmd.Flags().GetInt("direct-channels")
+	if err != nil || directChannels < 0 {
+		return errors.New("Invalid direct-channels parameter")
+	}
+	postsPerDirectChannel, err := cmd.Flags().GetInt("posts-per-direct-channel")
+	if err != nil || postsPerDirectChannel < 0 {
+		return errors.New("Invalid points-per-direct-channel parameter")
+	}
+	groupChannels, err := cmd.Flags().GetInt("group-channels")
+	if err != nil || groupChannels < 0 {
+		return errors.New("Invalid group-channels parameter")
+	}
+	postsPerGroupChannel, err := cmd.Flags().GetInt("posts-per-group-channel")
+	if err != nil || postsPerGroupChannel < 0 {
+		return errors.New("Invalid points-per-group-channel parameter")
+	}
+	workers, err := cmd.Flags().GetInt("workers")
+	if err != nil {
+		return errors.New("Invalid workers parameter")
+	}
+	profileImagesPath, err := cmd.Flags().GetString("profile-images")
+	if err != nil {
+		return errors.New("Invalid profile-images parameter")
+	}
+	profileImages := []string{}
+	if profileImagesPath != "" {
+		profileImagesStat, err := os.Stat(profileImagesPath)
+		if os.IsNotExist(err) {
+			return errors.New("Profile images folder doesn't exists.")
+		}
+		if !profileImagesStat.IsDir() {
+			return errors.New("profile-images parameters must be a folder path.")
+		}
+		profileImagesFiles, err := ioutil.ReadDir(profileImagesPath)
+		if err != nil {
+			return errors.New("Invalid profile-images parameter")
+		}
+		for _, profileImage := range profileImagesFiles {
+			profileImages = append(profileImages, path.Join(profileImagesPath, profileImage.Name()))
+		}
+		sort.Strings(profileImages)
+	}
+
+	if workers < 1 {
+		return errors.New("You must have at least one worker.")
+	}
+	if teamMemberships > teams {
+		return errors.New("You can't have more team memberships than teams.")
+	}
+	if channelMemberships > channelsPerTeam {
+		return errors.New("You can't have more channel memberships than channels per team.")
+	}
+
+	var bulkFile *os.File
+	switch bulk {
+	case "":
+		bulkFile, err = ioutil.TempFile("/tmp", "mattermost-sample-data-")
+		defer os.Remove(bulkFile.Name())
+		if err != nil {
+			return errors.New("Unable to open temporary file.")
+		}
+	case "-":
+		bulkFile = os.Stdout
```

Comment:
Is stdout clear of other stuff, like logging (does that all go safely to stderr)? If not, I'm not really sure this is useful.

Reply:
The answer is no, you will have log data + sampledata data. I can imagine me using it to play with the options, but I'm not sure is really useful.

Reply:
Right. I guess it does no harm to have it there, so +1 for just leaving it as is.